Update v3 documentation to reflect changes to Rename (#233)
laminas/laminas-filter: docs/book/v3/file.md

The following set of options are supported:

- `match` (string; default: `*`): File name pattern to check if this filter should be applied to the input.

> Read the [`fnmatch` documentation](https://www.php.net/manual/en/function.fnmatch.php) for more detail

Reviewer: ```suggestion
Read the [`fnmatch` documentation](https://www.php.net/manual/function.fnmatch.php) for more detail
```

Drops the `/en/` path segment - the PHP site will then auto-detect browser language

> - `target_directory` (string; default: `*`): Directory to move the file(s) to, if specified.

Reviewer: ```suggestion
- `target_directory` (string; default: `*`): Directory to move the file(s) to, if specified. The target directory must exist and be writable by the server process.
```

> - `rename_to` (string: default: `*`): Rename the file to this, if specified.

Reviewer: ```suggestion
- `rename_to` (string: default: `*`): Change the filename to the specified value. A value of `*` _(The default)_ preserves the existing file name.
```

- `overwrite` (boolean; default: `false`): Shall existing files be overwritten?
If the file is unable to be moved into the target path, a `Laminas\Filter\Exception\RuntimeException` will be thrown.
- `randomize` (boolean; default: `false`): Shall target files have a random postfix attached?
The random postfix will generated with `uniqid('_')` after the file name and before the extension.
For example, `file.txt` might be randomized to `file_4b3403665fea6.txt`.

> An array of option sets is also supported, where a single `Rename` filter instance can filter several files using different options.
> The options used for the filtered file will be matched from the `match` option in the options set.

Reviewer: ```suggestion
A list of "option sets" is also supported, where a single `Rename` filter instance can be configured to match and rename files using different options based on the received filename.
for example, a list such as
```php
[
  ['match' => '*.txt', 'target_directory' => '/tmp/text-files'],
  ['match' => '*.pdf', 'target_directory' => '/tmp/pdf-files'],
];
```

Would move matching files to alternative directories based on the received filename.

```

### Usage Examples

Move all filtered files to a different directory:

```php
// 'target' option is assumed if param is a string
$filter = new \Laminas\Filter\File\Rename(['target_directory' => '/tmp/']);
echo $filter->filter('./myfile.txt');
// File has been moved to '/tmp/myfile.txt'
```

Rename all filtered files to a new name:

```php
$filter = new \Laminas\Filter\File\Rename(['rename_to' => 'newfile.txt');
echo $filter->filter('./myfile.txt');
// File has been renamed to 'newfile.txt'
```

Move to a new path, and randomize file names:

```php
$filter = new \Laminas\Filter\File\Rename([

>     'match'    => '/tmp/newfile.txt',

Reviewer: ```suggestion
    'match'    => '/tmp/*.txt',
```

Serves as an obvious example of using `fnmatch` to match _any_ `.txt` file

```php
$filter = new \Laminas\Filter\File\Rename([
    [

>         'match'    => 'fileA.txt'
>         'target_directory' => '/dest1/',
>         'rename_to'    => 'newfileA.txt',

Reviewer: ```suggestion
        'match'    => '*.pdf'
        'target_directory' => '/pdf-files/',
        'rename_to'    => 'newfileA.pdf',
```

        'overwrite' => true,
    ],
    [

>         'match'    => 'fileB.txt'
>         'target_directory' => '/dest2/',
>         'rename_to'    => 'newfileB.txt',

Reviewer: ```suggestion
        'match'    => '*.txt'
        'target_directory' => '/text-files/',
        'rename_to'    => 'newfileB.txt',
```

Changes config to show the use-case of moving different types of file to different target directories
